core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java

```diff
@@ -462,13 +462,14 @@ protected double parseSpeed( String str )
     }
 
     /**
-     * This method parses a string ala "00:00" (hours and minutes) or "0:00:00" (days, hours and
-     * minutes).
+     * This method parses a string ala 'hh:mm', format for hours and minutes 'mm', 'hh:mm' or 'hh:mm:ss'
+     * FIXME: Add support for ISO_8601 
```

Comment:
In the meantime I have added support for ISO_8601 and intended to add it as separte pull request. I don't think that it is a good idea to just say that it is often filled out in a wrong way and even document in a comment a example of way with a wrong usage. First I would fix this way.
Shouldn't it be possible to add a simple plausability check based on the distance, when it comes to the decision if to consider the value of the duration tag? In case that the calulated speed is less then some value e.g. 0.1 km/h we could ignore it and use the calculated duration based on an the assumed ferry speed instead.